com.unity.multiplayer.mlapi/Tests/Editor/InternalMessageHandlerTests.cs

```diff
@@ -0,0 +1,387 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using MLAPI.Configuration;
+using MLAPI.Internal;
+using MLAPI.Messaging;
+using MLAPI.Messaging.Buffering;
+using MLAPI.Serialization;
+using MLAPI.Transports;
+using MLAPI.Transports.Tasks;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+using Object = UnityEngine.Object;
+
+namespace MLAPI.EditorTests
+{
+    public class InternalMessageHandlerTests
+    {
+        [Test]
+        public void MessageHandlerReceivedMessageServerClient()
```

Comment:
This is valuable and it's always nice to see test coverage added! I don't think you need to block the PR on this, but I'd encourage you to do some reading about AAA (arrange-act-assert) pattern testing and try to apply it going forward.

It takes some time to see what's being verified in this test, and how. Part of that is because it's so large (testing multiple things), but part of it is because the "assert" is happening inside of the "act".